tv/tv.cpp

```diff
@@ -288,19 +352,40 @@ struct TVPass final : public llvm::FunctionPass {
     }
 
     // Regenerate tgt because preprocessing may have changed it
-    I->second.fn = *llvm2alive(F, *TLI);
-    return false;
+    if (!parallel_tv)
+      I->second.fn = *llvm2alive(F, *TLI);
+
+  done:
+    if (parallel_tv) {
+      llvm_util_init.reset();
+      smt_init.reset();
+      parallel->finishChild();
+    } else {
+      return false;
+    }
   }
 
   bool doInitialization(llvm::Module &module) override {
     if (initialized++)
       return false;
 
+    if (parallel_tv_jobserver) {
+      parallel = new jobServer();
+    } else if (parallel_tv_unrestricted) {
+      parallel = new unrestricted();
+    }
+
+    if (parallel) {
+      if (parallel->init(max_subprocesses))
+        parallel_tv = true;
+      else
+        std::cerr << "WARNING: Parallel execution of Alive2 Clang plugin is "
+          "unavailable, sorry\n";
+    }
+
     fnsToVerify.insert(opt_funcs.begin(), opt_funcs.end());
 
     if (!report_dir_created && !opt_report_dir.empty()) {
-      static default_random_engine re;
-      static uniform_int_distribution<unsigned> rand;
       static bool seeded = false;
 
       if (!seeded) {
```

Comment:
you missed this bit of the seed while sharing the random_engine. Like this all processes will have the same seed and filenames will clash

Reply:
ah, good catch. I've been using the mode where it just overwrites files so never saw that happen.